packages/openapi-code-generator/src/typescript/common/schema-builders/joi-schema-builder.spec.ts

```diff
@@ -1294,8 +1347,7 @@ describe.each(
       await expect(execute({foo: "bla"})).rejects.toThrow('"bar" is required')
     })
 
-    // TODO: https://github.com/hapijs/joi/issues/3057
-    it.skip("can intersect unions", async () => {
+    it("can intersect unions", async () => {
```

Comment:
Was somewhat forced to fix this, as integration tests started outputting intersected unions due to the richer anyOf / oneOf / allOf support.

The "fix" is rather hacky, but seems to work well enough